src/x509.c

```diff
@@ -4767,68 +4767,112 @@ WOLFSSL_X509_NAME* wolfSSL_X509_get_subject_name(WOLFSSL_X509* cert)
 
 #if defined(OPENSSL_EXTRA) && (!defined(NO_SHA) || !defined(NO_SHA256))
 /******************************************************************************
-* wolfSSL_X509_subject_name_hash - compute the hash digest of the raw subject name
-* This function prefers SHA-1 (if available) for compatibility
+* wolfSSL_X509_subject_name_hash
+* wolfSSL_X509_issuer_name_hash
+* Compute the hash digest of the subject / issuer name.
+* These functions prefers SHA-1 (if available) for compatibility.
 *
 * RETURNS:
-* The beginning of the hash digest. Otherwise, returns zero.
+* The first 4 bytes of SHA hash in little endian order as unsigned long.
+* Otherwise, returns zero.
 * Note:
-* Returns a different hash value from OpenSSL's X509_subject_name_hash() API
-* depending on the subject name.
+* Returns the same hash value as OpenSSL's X509_X_name_hash() API
+* if SHA-1 support is compiled in.
```

Comment:
Let's add documentation saying that SHA-256 wil be used if SHA-1 isn't available.